test/Strategy/ScalarTypeStrategyTest.php

```diff
@@ -8,14 +8,20 @@
 use PHPUnit\Framework\Attributes\CoversClass;
 use PHPUnit\Framework\TestCase;
 
+use const PHP_FLOAT_EPSILON;
+
 #[CoversClass(ScalarTypeStrategy::class)]
 final class ScalarTypeStrategyTest extends TestCase
 {
     public function testHydrate(): void
     {
         $this->assertSame(123, ScalarTypeStrategy::createToInt()->hydrate('123', null));
         $this->assertNull(ScalarTypeStrategy::createToInt()->hydrate(null, null));
-        $this->assertSame(123.99, ScalarTypeStrategy::createToFloat()->hydrate('123.99', null));
+        $this->assertEqualsWithDelta(
+            123.99,
+            ScalarTypeStrategy::createToFloat()->hydrate('123.99', null),
+            PHP_FLOAT_EPSILON
+        );
```

Comment:
Not a fan of this rector. It seems unnecessary when `assertSame` passes and is IMO easier to read. Is it easy to skip?